noobaa/noobaa-core
# S3select parquet fs

### Explain the changes
1. Implement s3select for parquet in nsfs.


### Issues: Fixed #xxx / Gap #xxx
1. No support for other namespaces nor noobaa.
2. build issues TBD (specifically downstream, but upstream build changes are also not final yet)

### Testing Instructions:
1. Get a parquet file in nsfs. A simple way is with
-node src/core/nsfs.js /home/aprinzse/work/nsfs/
-alias s3-user1='AWS_ACCESS_KEY_ID=dc1 AWS_SECRET_ACCESS_KEY=dc2 aws --endpoint https://127.0.0.1:6443 --no-verify-ssl s3'
-s3-user1 mb s3://<bucket name>
-s3-user1 cp <parquet_file> s3://<bucket name>

-AWS_ACCESS_KEY_ID=dc1 AWS_SECRET_ACCESS_KEY=dc2 aws --endpoint https://127.0.0.1:6443 --no-verify-ssl s3api select-object-content --bucket <bucket name> --key <filename> --expression "select * from stdin;" --expression-type 'SQL' --input-serialization '{"Parquet": {}}' --output-serialization '{"CSV": {}}' "output.csv"
